PhysicsTools/PatAlgos/plugins/PATJetSelector.h

```diff
@@ -61,13 +64,20 @@ namespace pat {
       edm::Handle< edm::View<pat::Jet> > h_jets;
       iEvent.getByToken( srcToken_, h_jets );
 
+      unsigned nl = 0; // number of loose jets
       // First loop over the products and make the secondary output collections
       for ( edm::View<pat::Jet>::const_iterator ibegin = h_jets->begin(),
 	      iend = h_jets->end(), ijet = ibegin;
 	    ijet != iend; ++ijet ) {
 
 	// Check the selection
-	if ( selector_(*ijet) ) {
+	bool selectedLoose = false;
+	if ( nLoose_ > 0 && selectorLoose_(*ijet) ) {
```

Comment:
No reason to compute it when nl>nLoose:

unsigned nl = nLoose;  // number of remaining loose jets to consider
...
bool selectedLoose = false;
if (nl>0) { selectedLoose = true; --nl; }
bool pass = cut_(*it) || selectedLoose